File: solutions/c/two-fer/2/two_fer.c
#include "two_fer.h"

#include <stdio.h>
#include <string.h>

void two_fer(char *const buffer, const char *const name) {
    sprintf(buffer, "One for %s, one for me.", !name || strlen(name) == 0 ? "you" : name);
}

File: solutions/java/two-fer/1/src/main/java/Twofer.java
public class Twofer {
    public static String twofer(String name) {
        return switch (name) {
            case null -> DEFAULT_STRING;
            case "" -> DEFAULT_STRING;
            default -> "One for " + name + ", one for me.";
        };
    }

    private static String DEFAULT_STRING = "One for you, one for me.";
}